packages/world/ts/config/v2/world.ts

```diff
@@ -75,29 +74,42 @@ export function resolveWorld<const world extends WorldInput>(world: world): reso
   const scope = extendedScope(world);
   const namespaces = world.namespaces ?? {};
 
+  const resolvedStore = resolveStore(world);
+
+  const codegen = mergeIfUndefined(resolvedStore.codegen, resolveCodegen(world.codegen));
+  codegen.namespaceDirectories = world.namespaces !== undefined;
+  codegen.indexFilename = world.namespaces === undefined ? codegen.indexFilename : false;
```

Comment:
using both resolved store codegen and resolved world codegen is a bit weird since the resolvers will attempt to never leave something `undefined`, making it hard to set some defaults here

I think this is closer to accurate, where we set up some store codegen defaults if they're not already set:

```suggestion
  const resolvedStore = resolveStore({
    ...world,
    codegen: mergeIfUndefined(world.codegen ?? {}, {
      namespaceDirectories: world.namespaces != null,
      indexFilename: world.namespaces != null ? false : undefined,
    }),
  });
```